lib/bin/lwm2m_carrier/os/lwm2m_os.c

```diff
@@ -239,7 +239,7 @@ void lwm2m_os_timer_release(lwm2m_os_timer_t *timer)
 		return;
 	}
 
-	work->handler = NULL;
+	k_work_cancel_delayable_sync(&work->work_item, &work->work_sync);
```

Comment:
You should actually do both - cancel the work, and then release the timer, by setting its handler to NULL. Since the work canceling is synchronous, there's no risk it'd be exectued later.

Reply:
I'm afraid this will not work, because the Carrier lib reuses this work however it never reassigns the handler, so it stays NULL.

Reply:
Well, it's your library and it's up to you really how you want to resolve the issue, but... to me it just doesn't sound right to reimplement API function against its specification just to address API misuse in the library.

I understand that modifying carrier lib might not be possible, but perhaps it should be clearly marked as a workaround and addressed on the carrier library side in the future?

Reply:
ok, I've left the response comment after the comment was edited. I still think however that some comment with explaination could be left along with the change.

Reply:
I agree that the API has not been used properly. I'll add a comment to explain this workaround